결제 관련 개선점, 미니게임 추가

sngsngUDON opened the pull request:
## #️⃣연관된 이슈

> 

## 📝작업 내용

> 1. 친구 선택할 때 본인이 추가가 되지 않아서 필수적으로 추가되도록 수정했습니다.
> 2. 다같이 결제 창에서 미니게임과 연동하였습니다.


### 스크린샷 (선택)


## 💬리뷰 요구사항(선택)

>

sngsngUDON committed:
1차 개선점

1. 본인이 추가가 안된것을 확인하고 사용자 본인은 무조건 포함시키게 개선
2. 어차피 회사 사람들만 나오고 있어서 회사 탭을 제외하고 전체와 부서만 남김

sngsngUDON committed:
본인 필수 추가 및 미니게임 연결

ogh010 commented on src/main/java/kr/co/solfood/user/cart/CartController.java:
```diff
@@ -244,17 +245,36 @@ public ResponseEntity<Map<String, Object>> getSelectedFriends(HttpSession sessio
             List<UserVO> selectedFriends = new ArrayList<>();
```
Autowired로 안하구 new ArrayList<>() 로 한 이유가 뭔가요??

ogh010 commented on src/main/webapp/WEB-INF/views/user/cart/invite-friends.jsp:
```diff
@@ -224,10 +224,15 @@
     <meta name="contextPath" content="${pageContext.request.contextPath}">
     
-    <!-- 현재 사용자 부서 ID -->
+    <!-- 현재 사용자 정보 -->
```
이건 뭔가요?
data-current-user-

ogh010 commented on src/main/webapp/WEB-INF/views/user/cart/waiting-approval.jsp:
```diff
@@ -76,6 +82,17 @@
     <input type="hidden" id="friendCountData" value="${friendCount != null ? friendCount : 0}">
     <input type="hidden" id="miniGameMessageData" value="<c:out value='${miniGameMessage}' escapeXml='true'/>">
     
+    <!-- 현재 사용자 정보 -->
```
data- 이런건 뭔가요? input hidden 으로 데이터 바인딩 한것과 무슨 차이가 잇나요

5y1ee commented:
대충 봤는데 js에 로직이 너무 많습니다.
프론트에 있으면 안되는 로직도 너무 많아요. 특히 금액 결정하는거나, 요청을 보내는 사람 결정하는 정보 역시 url에 그대로 노출되는게 치명적입니다. 이건 병합은 하되 로직은 제가 다 수정해야겠습니다.

sngsngUDON commented:
네네 검토가 좀 부족한 부분이 있어서 내일 회의하고 개선해보겠습니다

sngsngUDON committed:
페이징 처리 개선 및 금액 계산,친구 초대 로직 개선

1. 페이징 처리와 장바구니에서 계속 새로고침이 되는 부분을 ajax 를 사용해 개선했습니다.
2. 친구 선택/초대 와 금액 계산 부분을 api 로 만들어 controller 로 옮겼습니다.
